sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -283,45 +286,79 @@ class RangeKeyScanStateEncoder(
   // If the value is not null, we write the null byte followed by the value.
   // Note that setting null for the index on the unsafeRow is not feasible as it would change
   // the sorting order on iteration.
+  // Also note that the same byte is used to indicate whether the value is negative or not.
```

Comment:
More importantly, I think we need to comment why we chose the byte values the way we did. We can't encode positive numbers with `0x00`, for example. It needs to be `0x01`.

Reply:
Done - added notes above